Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm

```diff
@@ -82,17 +140,40 @@
 #endif
     }
 }
+#endif
 
 void ProcessLauncher::launchProcess()
 {
     ASSERT(!m_xpcConnection);
 
-    const char* name = serviceName(m_launchOptions, m_client);
+#if USE(EXTENSIONKIT)
+    auto handler = [](ThreadSafeWeakPtr<ProcessLauncher> weakProcessLauncher, _SEExtensionProcess* process, const char* name, NSError* error)
+    {
+        if (!weakProcessLauncher.get())
+            return;
+        if (error)
+            NSLog(@"Error launching process %@ error %@", process, error);
+        callOnMainRunLoop([weakProcessLauncher = weakProcessLauncher, name = name, process = RetainPtr<_SEExtensionProcess>(process)] {
+            auto connection = [process makeLibXPCConnectionError:nil];
+            weakProcessLauncher.get()->m_xpcConnection = connection;
```

Comment:
You should use a local variable for the result of `weakProcessLauncher.get()`. Calling `weakProcessLauncher.get()` is as cheap as you might expect in this case. So calling it 3 times in a row is not ideal.

Reply:
That is a good point! Will fix in upcoming patch.